src/rabbitmq/rabbitmq.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import amqplib, { Channel } from 'amqplib/callback_api';
+import config from 'config';
+import { DataSourceDocument } from '../data-source.model';
+
+interface HarvestMessage {
+  orgId: string;
+  catalogId: string;
+  dataSourceType: string;
+}
+
+const { host = '', port = '', exchange = '' } = config.get('rabbitmq');
+
+let channel: Channel;
+
+amqplib.connect(`amqp://${host}:${port}`, (err, connection) => {
+  if (err) {
+    throw err;
+  }
+
+  connection.on('error', console.error);
+
+  connection.createChannel((err, ch) => {
+    if (err) {
+      throw err;
+    }
+
+    ch.assertExchange(exchange, 'fanout', { durable: false });
+    channel = ch;
+  });
+});
+
+export const publishDataSource = ({
+  dataSourceType = '',
+  publisherId = ''
+}: DataSourceDocument): void => {
+  const message: HarvestMessage = {
+    orgId: publisherId,
+    catalogId: publisherId,
+    dataSourceType: dataSourceType
+  };
+
+  const envelope = {
+    type: `${dataSourceType.toUpperCase()}_HARVEST_TRIGGER`,
+    payload: message
+  };
+
+  channel.publish(exchange, '', Buffer.from(JSON.stringify(envelope)));
+};
+
+process.on('exit', () => {
+  channel.close(console.log);
```

Comment:
skal du sjekke hvis channel er åpnet?

Reply:
fikset

Reply:
egentlig trenger vi ikke å sjekke det hvis vi thrower err. For hvis vi ikke thrower err, så må vi også gjøre samme check i publishDataSource() før channel.publish()